File: source/math/Triangle.java
package math;

import java.io.Serializable;

public class Triangle implements Serializable {
    public Vertex a;
    public Vertex b;
    public Vertex c;

    public Triangle(Vertex A, Vertex B, Vertex C) {
        a = A;
        b = B;
        c = C;
    }

    public Triangle(Triangle triangle) {
        a = new Vertex(triangle.a);
        b = new Vertex(triangle.b);
        c = new Vertex(triangle.c);
    }
}

File: source/math/Triangle.kt
package math

import java.io.Serializable

class Triangle : Serializable {
    var a: Vertex = Vertex()
    var b: Vertex = Vertex()
    var c: Vertex = Vertex()

    constructor()

    constructor(A: Vertex, B: Vertex, C: Vertex) {
        a = A
        b = B
        c = C
    }

    constructor(triangle: Triangle) {
        a = Vertex(triangle.a)
        b = Vertex(triangle.b)
        c = Vertex(triangle.c)
    }
}